### ISSUE TITLE
feat: filterJson mongodb

### ISSUE THREAD

**Issue by reslear** [prisma/prisma#10879]
from https://github.com/prisma/prisma/discussions/8452

- https://github.com/prisma/prisma/issues/7135
- https://github.com/prisma/prisma/issues/3219
- https://github.com/prisma/prisma/issues/6708

**Comment by janpio** [comment]
Can you edit your issue to contain a bit more context and text? It is currently really hard to parse when coming into the issue with no previous context. A few short sentences are totally enough.

**Comment by reslear** [comment]
example mongodb:

```json5
[
  {
    "store": {
      products: 1,
      materials: 5
    }
  },
  {
    "store": {
      products: 5,
      materials: 4
    }
  },
  {
    "store": {
      products: 20,
      materials: 4
    }
  },
]
```

```ts
db.collection.find({
  "store.products": { $gt: 4 }
})
```

```json5
[
  {
    "_id": ObjectId("5a934e000102030405000001"),
    "store": {
      "materials": 4,
      "products": 5
    }
  },
  {
    "_id": ObjectId("5a934e000102030405000002"),
    "store": {
      "materials": 4,
      "products": 20
    }
  }
]
```

play: https://mongoplayground.net/p/rBzX2IMHLI3

**Comment by cernec1999** [comment]
Hello! I have a similar use case where I'd like to be able to filter on unstructured JSON. This is syntax that works with other databases, but not with mongodb:

```
    const esObject = await prisma.eSObject.findMany({
        where: {
            data: {
                path: ['beans'],
                contains: 'cool'
            }
        }
    })
```

**Comment by LogansUA** [comment]
Hey, is there any chance of implementing this feature?
Currently, the `path ... contains/equals` syntax also doesn't work on `jsonb` field in PostgreSQL where data is an array. Example:
```json
[
  { "id": "some-uuid-1", "name": "foo" },
  { "id": "some-uuid-2", "name": "bar" }
]
```
Query
```ts
const data = await this.prisma.test.findMany({
  where: {
    data: {
       path: ['id'],
       equals: "some-uuid-2",
    },
  },
});
```
Result
```ts
[]
```
😞

**Comment by FarhanMS123** [comment]
Hi! I am looking forward for these feature? Is this already implemented or in dev version? I have for some reason has this type of data:

```json
{
    "_id": "generated-objectid",
    "owners": [
        { "user_id": "user1", "priv": "investor" },
        { "user_id": "user2", "priv": "investor" },
        { "user_id": "user3", "priv": "leader" },
    ],
}
```

I may want to do something like:
```ts
prisma.some_collection.findMany({
    where: {
        "owners": {
            "path": "owners.$.user_id",
            "equals": "user2",
        },
    },
});
```

While I use MongoDB, it did not fully support `type` to encourage JSON by philosophy of MongoDB and NoSQL. So, I also use prisma-json-types-generator.

If may, can prisma currently do this? Is there any dev, canary, or nightly version which already support this? Should I open new issues?

**Comment by FarhanMS123** [comment]
> Hi! I am looking forward for these feature? Is this already implemented or in dev version? I have for some reason has this type of data:
> If may, can prisma currently do this? Is there any dev, canary, or nightly version which already support this? Should I open new issues?

Sorry for incovenience, my problem is not really relatable. It is working in `5.4.2` with this code:

```ts
prisma.test_deep_object.findMany({
        where: {
            owners: {
                some: {
                    user_id: "user2",
                }
            }
        }
    }
```

**Comment by debasispanigrahi** [comment]
What is the progress regarding this issue?

**Comment by stephenasuncionDEV** [comment]
Also want to filter if one of the fields contains a string

**Comment by reslear** [comment]
![image](https://github.com/prisma/prisma/assets/12596485/be815083-71c6-4e24-9e61-29d8a8890040)

<details>
  <summary>Spoiler warning</summary>
  
**¯\_(ツ)_/¯**
  
</details>

**Comment by charanjit-singh** [comment]
![image](https://github.com/prisma/prisma/assets/23359602/d86e72d3-76cb-49ba-89a3-920f49a10bf1)

**Comment by charanjit-singh** [comment]
No updates!

**Comment by charanjit-singh** [comment]
Guess prisma is not for MongoDB, there are issues like this basic filtering, no support of partial indexes etc.

**Comment by JwLDevs** [comment]
Will this feature ever be added? I really need this and there's no workarounds at all other than just ditching JSON entirely for prisma relations
